compose/cli/main.py

```diff
@@ -1358,9 +1352,10 @@ def remove_container(force=False):
         if options['--rm']:
             project.client.remove_container(container.id, force=True, v=True)
 
-    environment_file = options.get('--env-file')
-    environment = Environment.from_env_file(project_dir, environment_file)
-    use_cli = not environment.get_boolean('COMPOSE_INTERACTIVE_NO_CLI')
+    environment_file = toplevel_options.get('--env-file')
+    toplevel_environment = Environment.from_env_file(project_dir, environment_file)
+    use_cli = not toplevel_environment.get_boolean('COMPOSE_INTERACTIVE_NO_CLI')
```

Comment:
Why not using `use_cli = not self.toplevel_environment.get_boolean('COMPOSE_INTERACTIVE_NO_CLI')` here too?

Reply:
This is a stand-alone function, not a method of `TopLevelCommand`.  Though it seems like it's only called in `TopLevelCommand.run`, so adding an argument to pass `toplevel_environment` in could work fine and remove this duplication.

Reply:
Added a commit to do this--replaced the `project_dir` argument (which was only used for this) with a `toplevel_environment` argument.

Reply:
Cool! I didn't have enough context in the browser based diff to see that.